patroni/api.py

```diff
@@ -180,6 +180,80 @@ def do_GET_config(self):
+        metrics.append("# TYPE patroni_version gauge")
+        padded_semver = ''.join([x.zfill(2) for x in patroni.version.split('.')])  # 2.0.2 => 020002
+        metrics.append("patroni_version {0}".format(padded_semver).replace('.', ''))
+
+        metrics.append("# HELP patroni_postgres_running Value is 1 if Postgres is running, 0 otherwise.")
+        metrics.append("# TYPE patroni_postgres_running gauge")
+        metrics.append("patroni_postgres_running {0}".format(int(postgres['state'] == 'running')))
+
+        metrics.append("# HELP patroni_postmaster_start_time Epoch seconds since Postgres started.")
+        metrics.append("# TYPE patroni_postmaster_start_time counter")
+        metrics.append("patroni_postmaster_start_time {0}".format(int(self.get_postgresql_start_time_epoch(False))))
+
+        metrics.append("# HELP patroni_master Value is 1 if this node is the leader, 0 otherwise.")
+        metrics.append("# TYPE patroni_master gauge")
+        metrics.append("patroni_master {0}".format(int(postgres['role'] == 'master')))
+
+        metrics.append("# HELP patroni_xlog_location Current location of the Postgres"
+                       " transaction log, 0 if this node is not the leader.")
+        metrics.append("# TYPE patroni_xlog_location counter")
+        metrics.append("patroni_xlog_location {0}".format(postgres.get('xlog', {}).get('location', 0)))
+
+        metrics.append("# HELP patroni_standby_leader Value is 1 if this node is the standby_leader, 0 otherwise.")
+        metrics.append("# TYPE patroni_standby_leader gauge")
+        metrics.append("patroni_standby_leader {0}".format(int(postgres['role'] == 'standby_leader')))
+
+        metrics.append("# HELP patroni_replica Value is 1 if this node is a replica, 0 otherwise.")
+        metrics.append("# TYPE patroni_replica gauge")
+        metrics.append("patroni_replica {0}".format(int(postgres['role'] == 'replica')))
+
+        metrics.append("# HELP patroni_xlog_received_location Current location of the received"
+                       " Postgres transaction log, 0 if this node is not a replica.")
+        metrics.append("# TYPE patroni_xlog_received_location counter")
+        metrics.append("patroni_xlog_received_location {0}".format(
+                        postgres.get('xlog', {}).get('received_location', 0)))
+
+        metrics.append("# HELP patroni_xlog_replayed_location Current location of the replayed"
+                       " Postgres transaction log, 0 if this node is not a replica.")
+        metrics.append("# TYPE patroni_xlog_replayed_location counter")
+        metrics.append("patroni_xlog_replayed_location {0}".format(
+                        postgres.get('xlog', {}).get('replayed_location', 0)))
+
+        metrics.append("# HELP patroni_xlog_replayed_timestamp Current timestamp of the replayed"
+                       " Postgres transaction log, 0 if null.")
+        metrics.append("# TYPE patroni_xlog_replayed_timestamp counter")
```

Comment:
This one is tricky. It could be the gauge as well as the counter.
The gauge - because it simply shows absolute value.
But, the counter could be also useful, because it makes it possible to estimate how fast it is catching up with the replication lag or how fast the replication lag is growing.

Reply:
Agree, I'm not sure the `TYPE` has an effect on anything other than humans (and potentially "help" inside of interfaces, e.g., I seem to recall that Grafana displays `HELP` and `TYPE` in the Data Explorer while a person is exploring their metrics). With that said, since it's "what it is" when it's scraped, `GAUGE` probably makes the most sense. As far as estimating how fast it's catching up, a person viewing the metric in a dashboard could get some sense by looking at the slope/trend of the curve over time.